### PR TITLE
GP2-1144: Allow Beta token checks to support sub-paths of allowed paths

### PR DESCRIPTION
**Pull request by stevejalim**
This changeset should reduce the noise in Sentry for paths which lack a Beta token but don't need the token anyway.

Plus, this fixes a broken UI issue on Beta:  `/static/` is currently configured as an allowed path, but only as a distinct, _complete_ path, rather than the head of a longer path to some static assets. This means that the login/landing page for a user on Beta who does NOT have a token lacks all the JS and CSS we'd want them to see. This bug should be fixed by these changes.

Adds specific tests, and not just to keep coverage levels up ;)

 - [X] Ticket exists in Jira  https://uktrade.atlassian.net/browse/TICKET_ID_HERE 
 - [X] Jira ticket has the correct status.
 - [X] [Changelog](CHANGELOG.md) entry added.
 
 - [ ] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: core/middleware.py [hunk 1]
```diff
@@ -99,10 +99,16 @@ def __init__(self, get_response):
...
         response = self.get_response(request)
-        # need to whitelist the endpoint, to be able to generate tokens
-        # == '/api/create-token/' or request.path == '/favicon.ico':
-        if request.path in settings.BETA_WHITELISTED_ENDPOINTS:
-            return response
+        # Need to whitelist the endpoint, to be able to generate tokens
+        # == '/api/create-token/' or request.path == '/favicon.ico'.
+        # Note that we now support sub-paths of the whitelisted endpoints,
+        # not just the absolute endpoint paths
+        for whitelisted_path in settings.BETA_WHITELISTED_ENDPOINTS.split(','):
```
Comment (stevejalim) [inline core/middleware.py:106]: Note the change here from substring checking to actually splitting the comma-separated list

Comment (stevejalim) [reply core/middleware.py:106]: Also, there is a risk, I think, of crafted URLs (eg starting with an allowed path) to bypass the token check eg `/favicon.ico/extra/path/here/` but I don't think this is likely to reveal much info as it'll just 404. 

Do reviewers agree?

Comment (agiamas) [reply core/middleware.py:106]: yes

**Review by agiamas — APPROVED** [review]
lgtm

File: core/middleware.py [hunk 2]
```diff
@@ -99,11 +99,20 @@ def __init__(self, get_response):
     def __call__(self, request):
 
         response = self.get_response(request)
-        # need to whitelist the endpoint, to be able to generate tokens
-        # == '/api/create-token/' or request.path == '/favicon.ico':
-        if request.path in settings.BETA_WHITELISTED_ENDPOINTS:
+
+        # Always allow the homepage to be accessed without a token
+        if request.path == '/':
             return response
```
Comment (stevejalim) [inline core/middleware.py:105]: @agiamas Updated to explicitly always allow `/` but not subpages of it (unless they are in the allowed list, of course)

**Review by agiamas — APPROVED** [review]
lgtm

File: config/env/test [hunk 3]
```diff
@@ -21,5 +21,5 @@ AIRTABLE_API_KEY=debug
 DIT_HELPDESK_URL=http://www.debug.com
 FEATURE_FLAG_HARD_CODE_USER_INDUSTRIES_EXPERTISE=false
 BETA_TOKEN=z7445AQ6aGXoPdFxYXQaJPLa3XU6I0zZUuI-Kbar8Jk=
-BETA_WHITELISTED_ENDPOINTS='/api/create-token/,/favicon.ico,/api/check/,/admin/,/auth/login/,/'
+BETA_WHITELISTED_ENDPOINTS='/api/create-token/,/favicon.ico,/api/check/,/admin/,/auth/login/'
```
Comment (stevejalim) [inline config/env/test:24]: Fixed here and in local dev conf

**Review by agiamas — APPROVED** [review]
lgtm

**Pull request merged**